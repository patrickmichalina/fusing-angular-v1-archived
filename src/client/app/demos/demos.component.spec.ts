import { DemosComponent } from './demos.component'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { Component } from '@angular/core'
import { DemosModule } from './demos.module'
import { AppTestingModule } from '../../../testing/app-testing.module'

@Component({
  selector: 'test-component',
  template: '<pm-demos></pm-demos>'
})
class TestComponent {}

describe(DemosComponent.name, () => {
  let fixture: ComponentFixture<TestComponent>

  beforeEach(
    async(() => {
      TestBed.configureTestingModule({
        imports: [AppTestingModule.forRoot(), DemosModule],
        declarations: [TestComponent]
      }).compileComponents()
    })
  )

  beforeEach(
    async(() => {
      fixture = TestBed.createComponent(TestComponent)
    })
  )

  afterEach(
    async(() => {
      TestBed.resetTestingModule()
    })
  )

  it(
    'should compile',
    async(() => {
      fixture.detectChanges()
      expect(fixture.nativeElement).toBeDefined()
      expect(fixture).toMatchSnapshot()
    })
  )
})
