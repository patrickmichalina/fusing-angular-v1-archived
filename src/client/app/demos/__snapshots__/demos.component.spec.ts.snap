// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DemosComponent should compile 1`] = `
<test-component>
  <pm-demos>
    <h1>
      Demos
    </h1>
    <p>
      Certain things can be difficult when trying to achieve pure isomorphism. This page showcases these solved issues.
    </p>
    <h3>
      Caching Server XHR Requests
    </h3>
    <p>
      Don't make a second API request when the data was retrieved from the server already. This produce "flicker-free" responses. Reload this page and pay attention to how the list below does not flicker.
    </p>
    
  </pm-demos>
</test-component>
`;
